### PR TITLE
rest action doc, device.name

### PR DESCRIPTION
**Pull request by KamyarZiabari**

### CODE REVIEW

File: develop-guide/writeblitz/writeblitz.rst [hunk 1]
```diff
@@ -246,6 +246,60 @@ other existing apis.
         function: get_traffic_stream_objects
         ...
 
+rest
+____
+
+The ``rest`` action allows you to make rest call to any endpoint on a device. Rest uses http protocols to 
+transfer data. Five http protocols are supported, `get`, `post`, `put`, `patch` and `delete`.
```
Comment (jeaubin) [inline develop-guide/writeblitz/writeblitz.rst:253]: http method

File: develop-guide/writeblitz/writeblitz.rst [hunk 2]
```diff
@@ -246,6 +246,60 @@ other existing apis.
         function: get_traffic_stream_objects
         ...
 
+rest
+____
+
+The ``rest`` action allows you to make rest call to any endpoint on a device. Rest uses http protocols to
```
Comment (jeaubin) [inline develop-guide/writeblitz/writeblitz.rst:252]: allows to make 

remove the you

File: develop-guide/writeblitz/writeblitz.rst [hunk 3]
```diff
@@ -246,6 +246,60 @@ other existing apis.
...
         ...
 
+rest
+____
+
+The ``rest`` action allows you to make rest call to any endpoint on a device. Rest uses http protocols to 
+transfer data. Five http protocols are supported, `get`, `post`, `put`, `patch` and `delete`.
+
+You can find additional information on rest, using this `tutorial
+<https://pubhub.devnetcloud.com/media/genie-feature-browser/docs/#/apis>`_.
```
Comment (jeaubin) [inline develop-guide/writeblitz/writeblitz.rst:256]: why linking to apis??

apis isnt a tutorial...

wrong link?

File: develop-guide/writeblitz/writeblitz.rst [hunk 4]
```diff
@@ -246,6 +246,60 @@ other existing apis.
...
+                            "id": "eth1/2",
+                            "userCfgdFlags": "admin_layer,admin_state"
+                          }
+                        ]
+                      }
+                    }
+                  }
+
+You can also change your connection alias from rest to anyone of the following 
+['cli', 'yang', 'rest', 'xml', 'web', 'restconf', 'netconf', 'gnmi'] and still take advantage
```
Comment (jeaubin) [inline develop-guide/writeblitz/writeblitz.rst:289]: ????

This only works for Rest...why are you bringing these?

File: develop-guide/writeblitz/writeblitz.rst [hunk 5]
```diff
@@ -246,6 +246,60 @@ other existing apis.
         function: get_traffic_stream_objects
         ...
 
+rest
+____
+
+The ``rest`` action allows you to make rest call to any endpoint on a device. Rest uses http protocols to 
+transfer data. Five http protocols are supported, `get`, `post`, `put`, `patch` and `delete`.
+
```
Comment (jeaubin) [inline develop-guide/writeblitz/writeblitz.rst:254]: we need to see whats the structure for each of them

How to use them

Where is the post

File: develop-guide/writeblitz/writeblitz.rst (hunk 3 above)
Comment (KamyarZiabari) [reply develop-guide/writeblitz/writeblitz.rst:256]: damn it wrong link

File: develop-guide/writeblitz/writeblitz.rst (hunk 4 above)
Comment (KamyarZiabari) [reply develop-guide/writeblitz/writeblitz.rst:289]: But then why do we set connection_alias in the code ?

File: develop-guide/writeblitz/writeblitz.rst (hunk 2 above)
Comment (tahigash) [inline develop-guide/writeblitz/writeblitz.rst:252]: Rest uses http/https

File: develop-guide/writeblitz/writeblitz.rst (hunk 5 above)
Comment (KamyarZiabari) [reply develop-guide/writeblitz/writeblitz.rst:254]: so should I just add example for the post and patch?

File: develop-guide/writeblitz/writeblitz.rst (hunk 2 above)
Comment (KamyarZiabari) [reply develop-guide/writeblitz/writeblitz.rst:252]: Done!

File: develop-guide/writeblitz/writeblitz.rst (hunk 1 above)
Comment (KamyarZiabari) [reply develop-guide/writeblitz/writeblitz.rst:253]: Done!

File: develop-guide/writeblitz/writeblitz.rst [hunk 6]
```diff
@@ -246,6 +246,60 @@ other existing apis.
...
+You can find additional information on rest, using this `tutorial
+<https://pubhub.devnetcloud.com/media/genie-feature-browser/docs/#/apis>`_.
+
+.. code-block:: YAML
+
+    test_sections:
+        - plain_actions:
+            - rest:
+                method: get
```
Comment (tahigash) [inline develop-guide/writeblitz/writeblitz.rst:263]: better to have all the examples for each method. `post` and `patch` are missing.

File: develop-guide/writeblitz/writeblitz.rst (hunk 4 above)
Comment (tahigash) [reply develop-guide/writeblitz/writeblitz.rst:289]: with cli, `still take advantage` what does it mean?

File: develop-guide/writeblitz/writeblitz.rst (hunk 2 above)
Comment (tahigash) [inline develop-guide/writeblitz/writeblitz.rst:252]: what are you referring to as `endpoint`?

File: develop-guide/writeblitz/writeblitz.rst [hunk 7]
```diff
@@ -246,6 +246,60 @@ other existing apis.
...
+['cli', 'yang', 'rest', 'xml', 'web', 'restconf', 'netconf', 'gnmi'] and still take advantage 
+of this action to transfer data from any endpoint that is available on the device.
+
+.. code-block:: YAML
+
+    test_sections:
+        - plain_actions:
+            - rest:
+                method: get
+                connection_alis: xml
```
Comment (tahigash) [inline develop-guide/writeblitz/writeblitz.rst:298]: this works?

File: develop-guide/writeblitz/writeblitz.rst (hunk 4 above)
Comment (jeaubin) [reply develop-guide/writeblitz/writeblitz.rst:289]: cause you can use any custom one they want,   but these are mapping datafile.

File: develop-guide/writeblitz/writeblitz.rst (hunk 5 above)
Comment (jeaubin) [reply develop-guide/writeblitz/writeblitz.rst:254]: yes

File: develop-guide/writeblitz/writeblitz.rst [hunk 8]
```diff
@@ -627,6 +681,12 @@ in action ``execute`` output and print the   :monospace:`main_learn_output` into
     Both filter and include/exclude features are using our dictionary querying tool `Dq
     <https://pubhub.devnetcloud.com/media/genie-docs/docs/userguide/utils/index.html#dq>`_.
 
+.. note::
+
+    The name of the device that the action is being executed on will be saved automatically upon
+    execution of the action and stay usable till the end of that action lifecycle. You can use that 
+    name as a variable using ``VARIABLES{device.name}`` for various purposes in your action.
```
Comment (tahigash) [inline develop-guide/writeblitz/writeblitz.rst:688]: missing `%`? `%VARIABLES{device.name}`. I think more friendly if we have an example of yaml.

File: develop-guide/writeblitz/writeblitz.rst [hunk 9]
```diff
@@ -751,6 +811,7 @@ Trigger timeout/interval ratio adjustments
 
 Each action performs verification to make sure it has performed as expected.
 These timeouts can be modified with a ratio from the testbed datafile.
+Actions ``api``, ``execute``, ``parse``, ``learn`` and ``rest`` are supporting this feature
```
Comment (tahigash) [inline develop-guide/writeblitz/writeblitz.rst:814]: This feature is supported by actions `api`,`execute`, `parse`, `learn` and `rest`.

above is better?

Comment (KamyarZiabari) [reply develop-guide/writeblitz/writeblitz.rst:834]: DOne

File: develop-guide/writeblitz/writeblitz.rst (hunk 8 above)
Comment (KamyarZiabari) [reply develop-guide/writeblitz/writeblitz.rst:708]: Done

File: develop-guide/writeblitz/writeblitz.rst (hunk 2 above)
Comment (KamyarZiabari) [reply develop-guide/writeblitz/writeblitz.rst]: uri rest endpoint si 1.1.1.1/api/mo/sys/bgp.json. From after ip that is called endpoint

File: develop-guide/writeblitz/writeblitz.rst (hunk 6 above)
Comment (KamyarZiabari) [reply develop-guide/writeblitz/writeblitz.rst:263]: Added

File: develop-guide/writeblitz/writeblitz.rst (hunk 2 above)
Comment (KamyarZiabari) [reply develop-guide/writeblitz/writeblitz.rst]: it does but, get, post, put etc are known as http methods, and https is kind of implied when you say http methods

Comment (KamyarZiabari) [reply develop-guide/writeblitz/writeblitz.rst]: And for with cli, still take advantage what does it mean? I deleted that section, it seemed unnecessary to write that part

File: develop-guide/writeblitz/writeblitz.rst (hunk 7 above)
Comment (KamyarZiabari) [reply develop-guide/writeblitz/writeblitz.rst]: deleted that instance

**Pull request merged**